akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterActorSetImpl.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.internal
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.Terminated
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.ClusterShardingSettings.StateStoreModeDData
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl.EntityParent
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl.KeepAlivePinger
+import akka.cluster.sharding.typed.scaladsl.ClusterActorSet
+import akka.cluster.sharding.typed.scaladsl.ClusterActorSetSettings
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.sharding.typed.scaladsl.EntityContext
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+
+import scala.concurrent.duration.Duration
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object ClusterActorSetImpl {
+  object EntityParent {
+    sealed trait Command
+    case object Ping extends Command
```

Comment:
I just realized it could perhaps be more efficient to use the `akka.cluster.sharding.Shard.RestartEntity` command. It wouldn't allow us to do the ping-pong mentioned earlier though.